[염시온]-vending-machine

sionyeom opened the pull request:
- [x]  투입 금액
    - [x]  양수만 입력 가능
    - [x]  투입 후 투입금액 빈칸 변경
- [x]  상품
    - [x]  버튼 금액 300~1100원, 100원 단위 증가
    - [x]  상품 이름 금액 포함형태 표시(ex:FE300)
    - [x]  세가지 상태 표현(normal, hover, active)
- [x]  금액 표시창
    - [x]  숫자 중간 정렬
    - [x]  세자리마다 쉼표(,) 표시
    - [x]  버튼 상단 금액 표시창의 기본값 0 고정
    - [x]  금액이 없는 상태에서 상품 클릭시, 해당 상품 금액 노출
- [x]  금액 투입 입력칸 하단 배치 로그창
    - [x]  모든 동작 기록
    - [x]  아래 있을 수록 최신 로그
- [x]  예외
    - [x]  구매 후
        - [x]  상품 최소가보다 적을시, 금액 자동 반환

sionyeom committed:
Feat: 개발환경 세팅

1. node 개발환경 세팅
2. index.html 작성
3. tailwindcss 설치 및 적용

sionyeom committed:
Feat: html 퍼블리싱 완료

sionyeom committed:
Refactor: HTML 퍼블리싱 리팩토링

1. id 수정
2. item 생성 함수명 수정
3. item 생성 함수 선언부 수정

sionyeom committed:
Feat: 잔금 데이터 세팅 기능 추가

sionyeom committed:
Feat: 로그 생성 함수 생성

sionyeom committed:
Feat: 반환 기능 추가

sionyeom committed:
Feat: 구매 기능 추가

sionyeom committed:
Feat: tailwind css 추가

AyoungWon commented on script.js:
```diff
@@ -0,0 +1,129 @@
+}
+
+function purchaseItem() {
+  let itemValue = this.getAttribute("value");
+  let balanceDiv = document.getElementById("balance");
+  let balance = localStorage.getItem("balance");
+  let calBalance = Number(balance) - Number(itemValue);
```
const말고 let으로 하신 이유가 있을까요? 값이 변하는 경우가 없어보여서요 🤔

sionyeom replied:
말씀하신대로 값이 변하지 않기 때문에 const로 수정하는게 좋을 것 같네요! 피드백 감사합니다!

Gentlee-dev commented on script.js:
```diff
@@ -0,0 +1,129 @@
+
+  insertBtn.addEventListener("click", function () {
+    let amountValue = getItemValue();
+    if (amountValue > 1) {
```
1원은 투입하지 못하는걸까요ㅠㅠ

sionyeom replied:
엇.. 해당 1원 이상 투입이 가능해야하는데 수정해야겠네요. 감사합니다ㅎㅎ

Gentlee-dev commented on script.js:
```diff
@@ -0,0 +1,129 @@
+
+function getItemValue() {
+  let { value } = document.getElementById("input");
+  document.getElementById("input").value = "";
```
getItemValue함수명으로 보아
투입하고자 하는 값을 반환하는 함수로 보입니다.
값을 초기화하는 역할이 같이 있는 이유가 있을까요?

sionyeom replied:
투입 후 값을 초기화하려고 했는데 말씀하신대로 함수명에 적합한 기능이 아닌 것으로 생각이 되네요 감사합니다!

Gentlee-dev commented on script.js:
```diff
@@ -0,0 +1,129 @@
+  } else {
+    addLog("잔액이 부족합니다.");
+
+    if (Number(balance) === 0) {
```
구매하려는 상품의 금액이
잔액보다 높을 경우여서
조건문 수정해주셔도 좋을거같습니다.

sionyeom replied:
말씀하신 조건이 좀 더 명확한 조건이 되겠네요! 감사합니다!

Gentlee-dev commented on script.js:
```diff
@@ -0,0 +1,129 @@
+      balanceDiv.textContent = itemValue;
+      setTimeout(() => {
+        balanceDiv.textContent = "0";
+      }, 1000);
```
마우스로 누르는 동안 일어나는 이벤트여서
mousedown같은 이벤트 찾아보시면 도움 될 수 있을거같아요

sionyeom replied:
아하 누르는 동안 발생하는 이벤트였군요! 레퍼런스도 같이 공유해주셔서 감사합니다.

좋은 피드백 많이 주셔서 감사합니다 정중님 :blush:

sionyeom committed:
Refactor: 피드백 적용

1. 선언자 변경

sionyeom committed:
Fix: 기능 피드백 적용

1. 마우스 클릭 후 제거시 잔액 변경 기능 수정
2. 투입 금액 조건문 수정
3. 구입 금액 조건문 수정

sionyeom committed:
Refactor: 피드백 적용

1. 기능별로 파일 분리
2. 이벤트 핸들러 활용하여 this 사용 최소화
3. HTML 시멘틱 태그로 수정
4. form 활용
5. 뷰와 로직 분리
